packages/@aws-cdk/aws-stepfunctions-tasks/lib/codebuild/batch-get-reports.ts

```diff
@@ -0,0 +1,56 @@
+import * as iam from '@aws-cdk/aws-iam';
+import * as sfn from '@aws-cdk/aws-stepfunctions';
+import { Construct } from 'constructs';
+import { integrationResourceArn, validatePatternSupported } from '../private/task-utils';
+
+/**
+ * Properties for CodeBuildBatchGetReports
+ */
+export interface CodeBuildBatchGetReportsProps extends sfn.TaskStateBaseProps {
+  /**
+   * An array of ARNs that identify the Report objects to return.
+   */
+  readonly reportArns: string[];
+}
+
+/**
+ * Return an array of reports as a task
+ *
+ * @see https://docs.aws.amazon.com/step-functions/latest/dg/connect-codebuild.html
+ */
+export class CodeBuildBatchGetReports extends sfn.TaskStateBase {
+  private static readonly SUPPORTED_INTEGRATION_PATTERNS: sfn.IntegrationPattern[] = [
+    sfn.IntegrationPattern.REQUEST_RESPONSE,
+  ];
+
+  protected readonly taskMetrics?: sfn.TaskMetricsConfig;
+  protected readonly taskPolicies?: iam.PolicyStatement[];
+
+  private readonly integrationPattern: sfn.IntegrationPattern;
+
+  constructor(scope: Construct, id: string, private readonly props: CodeBuildBatchGetReportsProps) {
+    super(scope, id, props);
+    this.integrationPattern = props.integrationPattern ?? sfn.IntegrationPattern.REQUEST_RESPONSE;
+
+    validatePatternSupported(this.integrationPattern, CodeBuildBatchGetReports.SUPPORTED_INTEGRATION_PATTERNS);
+
+    this.taskPolicies = [
+      new iam.PolicyStatement({
+        resources: ['*'],
```

Comment:
same comment, getting reports needs to be scoped to a project.

Reply:
BatchGetReports is a bit different it takes in a reportArn as input: for example ```arn:aws:codebuild:<region>:<AccountId>:report/CodeBuildProject-pUVXud5zwfgJ-emptyReport:f8bd8423-0bda-46f8-af96-e477eb5b5be3```
IAM permissions are grouped around reportNames which is defined in buildSpec, something we can't easily extract from IProject. (@skinny85 please correct me if I'm wrong).

Options are rather limited here
Option 1: maintain the request input. We treat both static and dynamic input equally with ```arn:aws:codebuild:[[region]]:*:report-group/*``` which is used by SFN's dynamic template
Option 2:  Implement a policy analyzer, base on the input first determine if the reportArns are passed in dynamically or statically, if it's passed in dynamically we use arn:aws:codebuild:[[region]]:*:report-group/* otherwise we analyze each reportArns, validate the arn, then parse the arn and generate the policy. 

For simplicity I'm leaning towards Option 1 for this API

Reply:
Same solution as replied in the thread above